soc/silabs/silabs_siwx91x/CMakeLists.txt

```diff
@@ -12,3 +14,31 @@ set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
     ${KERNEL_BIN_NAME}
     ${KERNEL_BIN_NAME}.rps
 )
+
+# Custom signing with Silicon Labs Commander
+if(CONFIG_SiWX91X_SILABS_COMMANDER_SIGN)
+    # Define the west sign command to run along with west build command
+    set(WEST_SIGN_CMD
+        ${CMAKE_COMMAND} -E env "PATH=$ENV{PATH}:$ENV{WEST_ROOT}/bin" "ZEPHYR_WORKSPACE_ROOT=${CMAKE_CURRENT_SOURCE_DIR}"
+        ${WEST} sign
+        -t silabs_commander
+        -d ${CMAKE_BINARY_DIR}
+        --sbin ${PROJECT_BINARY_DIR}/${KERNEL_BIN_NAME}.rps
+    )
+
+    # Add custom command to invoke west sign during build
+    add_custom_command(
+        OUTPUT ${KERNEL_BIN_NAME}.rps
+        COMMAND ${CMAKE_COMMAND} -E echo "Forcing execution in ${PROJECT_BINARY_DIR}"
+        COMMAND ${WEST_SIGN_CMD}
+        DEPENDS zephyr_final
+        VERBATIM
+    )
+
+    # Add custom target to ensure signing is part of the build
+    add_custom_target(silabs_commander_sign
+        ALL
+        DEPENDS ${KERNEL_BIN_NAME}.rps
+        COMMENT "Ensuring signing with Silicon Labs Commander is complete"
+    )
```

Comment:
I believe this target is useless (the user can run `west sign`). However, you can add some entries to `extra_post_build_commands` to launch the signature automatically with `west build`.